src/app/surveys/surveys.component.ts

```diff
@@ -34,8 +35,12 @@ export class SurveysComponent implements OnInit, AfterViewInit {
 
   ngOnInit() {
     this.surveys.filterPredicate = filterSpecificFields([ 'name' ]);
-    this.getSurveys().subscribe((surveys) => {
-      this.surveys.data = surveys;
+    this.getSurveys()
+    .subscribe((surveys) => {
+      this.getSubmission().subscribe((submissions) => {
```

Comment:
We should not nest observables.  Please use switchMap or forkJoin to flatten this code.